index.js

```diff
@@ -175,7 +175,11 @@ var isAddress = exports.isAddress = function (data) {
 // extraction of a signing key from the address.
 exports.getKeyFromAddress = function (addr) {
   if (addr.key) return addr.key
-  var data = MultiServerAddress.decode(addr)
+  try {
+    var data = MultiServerAddress.decode(addr)
+  } catch (err) {
+    console.error(new Error('Attempted connection with malformed multiserver-address ' + addr))
+  }
   if (!data) return
```

Comment:
@mixmix thanks much for the notes. are you wanting this to explicitly say `if (!data) return null`? I'm also looking for other places this method get's used (I did not see it in `ssb-invites`). The only place I could find it used was in [ssb-gossip](https://github.com/ssbc/ssb-gossip/blob/04b17c781b983980b318d9a0701060d0f831f7a7/index.js#L246). In the `ssb-gossip` example, returning undefined or null would be inline with what that module expects. I will keep looking for more references as well.

Reply:
Highly recommend using rg (ripgrep) and pointing it at the node_modules of a project like patchwork to find occurances of a method

Reply:
I don't know what it should return, but I would recommend making it explicit.
Right not it implicitly returns undefined.

Reply:
used `ripgrep` against patchbay, patchwork, patchfoo, patchfox, and oasis to get a sampling of usage. there were four underlying modules where this is used. 
- ssb-conn-hub
- ssb-lan
- ssb-gossip
- ssb-legacy-conn

`ssb-conn-hub` checks if a multiserver address is valid before passing an address to `ref.getKeysFromAddress()` so it wouldn't be impacted by this change. `ssb-lan` performs a truthiness check before using the returned value, so returning `null` or `undefined` would be OK. Still need to check the other two.

Reply:
hi @mixmix , `ssb-gossip` and `ssb-legacy-conn` both use ref.isAddress() to check for a valid address before calling getKeyFromAddress, so I think these would be OK with the change.

The code has been updated to return `undefined`. I've seen [other code](https://github.com/staltz/ssb-conn-hub/blob/master/src/index.ts#L86) that does type checks for undefined, so maybe this is what's assumed. A note has also been added to the README, stating that `getKeyFromAddress` returns `undefined` when it fails.